packs/elastic/actions/lib/static_metagen.py

```diff
@@ -0,0 +1,122 @@
+            "master_only": {
+                "description": "Only operate on elected master node.",
+                "type": "boolean",
+                "default": False
+            },
+            "timeout": {
+                "description": "Elasticsearch operation timeout in seconds.",
+                "type": "integer",
+                "default": 600
+            },
+            "log_level": {
+                "description": "Log level [critical|error|warning|info|debug].",
+                "type": "string",
+                "default": "warn"
+            },
+            "dry_run": {
+                "description": "Do not perform any changes.",
+                "type": "boolean",
+                "default": False
+            }
+        },
+        "runner_type": "run-python",
+        "description": "Run a Meta Action through a generic Runner.",
+        "enabled": True,
+        "entry_point": "curator_runner.py"}
+
+    parameter_meta = {
+        "type": "string"
+    }
+
+    def __init__(self, action_meta=None):
+        self.action_meta = StaticMetagen.action_meta
+        if action_meta is not None:
+            self.action_meta.update(action_meta)
+
+
+    def generate_action(self, module_type, action):
+        manifest = self.action_meta
+        manifest['name'] = "{0}_{1}".format(module_type, action)
+        manifest['parameters']['action']['default'] = action
+
+        fh = open('{0}_{1}.yaml'.format(module_type, action), 'w')
+        fh.write('---\n')
+        fh.write(yaml.dump(manifest, default_flow_style=False))
+        fh.close()
+
+
+    def generate_from_file(self, meta_file):
+        if meta_file is None:
+            return None
+        fh = open(meta_file)
+        actions = yaml.load(fh.read())
+        for manifest in self._merge_actions(actions):
+            fh = open('{0}.yaml'.format(manifest['name']), 'w')
+            fh.write('---\n')
+            fh.write(yaml.dump(manifest, default_flow_style=False))
+            fh.close()
+
+
+    def _merge_actions(self, actions):
+        for i, action in enumerate(actions):
+            name, meta = action.items()[0]
+            manifest = self.action_meta.copy()
+            manifest['name'] = name
+            manifest['parameters']['action']['default'] = name
+
+            for k, v in meta['parameters'].items():
+                nv = StaticMetagen.parameter_meta.copy()
+                nv.update(v); meta['parameters'][k] = nv
+
+            parameters = manifest['parameters'].copy()
+            parameters.update(meta['parameters'])
+            meta['parameters'] = parameters
+            manifest.update(meta)
+
+            if 'alias' in manifest:
+                alias_name = manifest.pop('alias')
+                alias_manifest = manifest.copy()
+                alias_manifest['name'] = alias_name
+                yield alias_manifest
+            yield manifest
+
+
+metagen = StaticMetagen()
+metagen.generate_from_file('lib/curator_actions.yaml')
```

Comment:
Nice, I like that.

It would also be good to add a section for authors / developers to the readme which documents what this script does and how to use it to generate metadata files.